tests/e2e/ctl_v3_member_test.go

```diff
@@ -230,31 +235,140 @@ func ctlV3MemberUpdate(cx ctlCtx, memberID, peerURL string) error {
 	return e2e.SpawnWithExpectWithEnv(cmdArgs, cx.envMap, " updated in cluster ")
 }
 
+// TestCtlV3PromotingLearner tests whether etcd can automatically fix the
+// issue caused by https://github.com/etcd-io/etcd/issues/19557.
 func TestCtlV3PromotingLearner(t *testing.T) {
-	e2e.BeforeTest(t)
+	testCases := []struct {
+		name             string
+		snapshotCount    int
+		promotionSuccess bool
+	}{
+		{
+			name:          "create snapshot after learner promotion and not saved to v3store",
+			snapshotCount: 10,
+		},
+		{
+			name:          "create snapshot before learner promotion and not saved to v3store",
```

Comment:
Wouldn't snapshot count set to zero mean we run with default (10`000), which would mean no snapshots at all? 

Don't remember if etcd makes a snapshot at the start, we were discussing changing it, but don't remember the conclusion.

Reply:
> Wouldn't snapshot count set to zero mean we run with default (10`000), which would mean no snapshots at all?

Yes, it's true, no snapshot at all. Updated the sub test name to avoid confusion.